src/components/OpenFileDialog/OpenFileDialog.js

```diff
@@ -0,0 +1,444 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    Box,
+    Modal,
+    ModalTitle,
+    ModalContent,
+    DataTable,
+    DataTableHead,
+    DataTableBody,
+    DataTableRow,
+    DataTableCell,
+    DataTableColumnHeader,
+    DataTableToolbar,
+    NoticeBox,
+    CircularLoader,
+    Button,
+} from '@dhis2/ui'
+import isEqual from 'lodash/isEqual'
+import PropTypes from 'prop-types'
+import React, { useEffect, useMemo, useState } from 'react'
+import { FileList } from './FileList'
+import { NameFilter } from './NameFilter'
+import { styles } from './OpenFileDialog.styles'
+import { OwnerFilter } from './OwnerFilter'
+import { PaginationControls } from './PaginationControls'
+import { VisTypeFilter } from './VisTypeFilter'
+
+const getResourceFromType = type => `${type}s`
+
+const getQuery = type => ({
+    resource: getResourceFromType(type),
+    params: ({
+        sortField = 'name',
+        sortDirection = 'asc',
+        page = 1,
+        filters,
+    }) => {
+        const queryParams = {
+            filter: filters,
+            fields: `id,type,displayName,title,displayDescription,created,lastUpdated,user,access,href`,
+            paging: true,
+            pageSize: 8,
+            page,
+        }
+
+        if (sortDirection !== 'default') {
+            queryParams.order = `${sortField}:${sortDirection}`
+        }
+
+        return queryParams
+    },
+})
+
+export const OpenFileDialog = ({
+    type,
+    open,
+    onClose,
+    onFileSelect,
+    onNew,
+    currentUser,
+}) => {
+    const filesQuery = useMemo(() => getQuery(type), [])
+    const defaultFilters = {
+        searchTerm: '',
+        createdBy: 'all',
+        visType: 'all',
+    }
+
+    const [{ sortField, sortDirection }, setSorting] = useState({
+        sortField: 'displayName',
+        sortDirection: 'asc',
+    })
+    const [page, setPage] = useState(1)
+    const [searchTerm, setSearchTerm] = useState(defaultFilters.searchTerm)
+    const [searchTimeout, setSearchTimeout] = useState(null)
+    const [filters, setFilters] = useState(defaultFilters)
+
+    const formatFilters = ({ searchTerm, createdBy, visType }) => {
+        const queryFilters = []
+
+        switch (createdBy) {
+            case 'byothers':
```

Comment:
why not camel case `byOthers`?

Reply:
Refactored this using constants for the select options, exported from the `CreatedByFilter` component (renamed from `OwnerFilter`).